### PR TITLE
Ivanchuk/Child refactor

### PR DESCRIPTION
**Pull request by v-ivanchuk**
Task #153

### CODE REVIEW

File: OutOfSchool/OutOfSchool.WebApi/Controllers/ChildController.cs [hunk 1]
```diff
@@ -67,6 +68,29 @@ public async Task<IActionResult> GetById(long id)
...
+        /// <returns>List of all children.</returns>
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetByParentId(long id)
+        {
+            this.ValidateId(id, localizer);
+
+            var children = await service.GetAllByParent(id).ConfigureAwait(false);
```
Comment (OlhaHoliak) [inline OutOfSchool/OutOfSchool.WebApi/Controllers/ChildController.cs:84]: I think we need to implement checkout whether ParentId belongs to User, who called method.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ChildControllerTests.cs [hunk 2]
```diff
@@ -92,6 +92,46 @@ public async Task GetChildById_WhenIdIsNotValid_ShouldReturnNull(long id)
...
+        {
+            // Arrange
+            service.Setup(x => x.GetAllByParent(id)).ReturnsAsync(children.Where(p => p.ParentId == id));
+
+            // Act
+            var result = await controller.GetByParentId(id).ConfigureAwait(false) as OkObjectResult;
+
+            // Assert
+            Assert.That(result, Is.Not.Null);
+            Assert.AreEqual(200, result.StatusCode);
```
Comment (yfedo) [inline OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ChildControllerTests.cs:107]: would be nice to use constants for 200 and 204 status code))
cold use this  Assert.IsNotNull(result); for isnot null check))

**Review by yfedo — APPROVED** [review]

File: OutOfSchool/OutOfSchool.WebApi/Controllers/ChildController.cs (hunk 1 above)
Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi/Controllers/ChildController.cs]: Dear @OlhaHoliak,
Thank you a lot for your advice. I think it is a good idea, so I have implemented it.
Please review.
Thank you a lot.

File: OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ChildControllerTests.cs (hunk 2 above)
Comment (v-ivanchuk) [reply OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ChildControllerTests.cs]: Dear @yfedo,
Thank you a lot for your advice. I have implemented them in the current PR.

**Review by OlhaHoliak — APPROVED** [review]

**Pull request merged**